src/main/kotlin/io/emeraldpay/dshackle/upstream/grpc/BitcoinGrpcUpstream.kt

```diff
@@ -149,9 +149,11 @@ class BitcoinGrpcUpstream(
     override fun stop() {
     }
 
-    override fun update(conf: BlockchainOuterClass.DescribeChain) {
-        upstreamStatus.update(conf)
-        this.capabilities = RemoteCapabilities.extract(conf)
+    override fun update(conf: BlockchainOuterClass.DescribeChain): Boolean {
+        val newCapabilities = RemoteCapabilities.extract(conf)
         conf.status?.let { status -> onStatus(status) }
+        return upstreamStatus.update(conf) || (newCapabilities != capabilities).also {
```

Comment:
`upstreamStatus.update(conf) || (newCapabilities != capabilities).also` - `also` won't work if `upstreamStatus.update(conf)` is true because the second statement of boolean expression won't be executed.
`(upstreamStatus.update(conf) || (newCapabilities != capabilities)).also` will be work

Reply:
готово